src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -211,8 +211,9 @@ ClassLoaderData* SystemDictionary::register_loader(Handle class_loader, bool cre
     if (class_loader() == nullptr) {
       return ClassLoaderData::the_null_class_loader_data();
     } else {
-      ClassLoaderData* cld = ClassLoaderDataGraph::find_or_create(class_loader);
-      if (Arguments::enable_preview() && EnableValhalla) {
+      bool created = false;
+      ClassLoaderData* cld = ClassLoaderDataGraph::find_or_create(class_loader, created);
+      if (created && Arguments::enable_preview() && EnableValhalla) {
```

Comment:
I'd like to get rid of the `EnableValhalla` while we are at it, `--enable-preview` implies `EnableValhalla`, plus this would mean one less place to remove the flag later.